### PR TITLE
DEPS: `array-api-strict` to v2 and array_api tests fix

### PR DESCRIPTION
**Pull request by samir-nasibli**
### Description

Trying validate #1902 with #1861 changes. In case of green CI, #1902 will be approved and merged and #1861 will be rebased for the new changes.

Based on #1861 
### TODO:
Additionally `array-api-compat` addition into requirements list will be checked

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Comment by samir-nasibli** [comment]
This validation raised some issues for some tests suits at `sklearnex/tests/test_run_to_run_stability.py`.
Not related with #1861 itself, just the test suits should be updated for the array_api inputs with `array-api-strict` v2.
Will close #1902 after this PR CI checks being green.
